tools/read-dependencies

```diff
@@ -49,8 +49,8 @@ MAYBE_RELIABLE_YUM_INSTALL = [
         grep -q "^proxy=" /etc/yum.conf || return 0
         error ":: http proxy in use => forcing the use of fixed URLs in /etc/yum.repos.d/*.repo"
         sed -i --regexp-extended '/^#baseurl=/s/#// ; /^(mirrorlist|metalink)=/s/^/#/' /etc/yum.repos.d/*.repo
-        sed -i 's/download\.fedoraproject\.org/dl.fedoraproject.org/g' /etc/yum.repos.d/*.repo
-        sed -i 's/download\.example/dl.fedoraproject.org/g' /etc/yum.repos.d/*.repo
+        sed -i 's/download.fedoraproject.org/dl.fedoraproject.org/g' /etc/yum.repos.d/*.repo
```

Comment:
Rather than remove the `\`, shouldn't these be `\\.`? We want to replace the literal `.`, but in sed, the `.` is any character, so it needs to be escaped as `\.`, but since it lives in a python string, we need to escape the `\` in the python string, thus `\\.`.

Or I think the `r` prefix would work here too.

Reply:
Let's escape the backslash since we are nested withing triple quotes and a sed single quote piped into a `sh -c` command. Tested this replacement works with http_proxy set and `./tools/run-container --package images:fedora/41 --keep`